updates/1.138/230703_forumthread_lock.xml

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<update xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.ofw.fi/xml/2011/java-xmldb-updater/UpdaterSchema.xsd">
+
+  <sql>alter table forumthread add lockBy bigint(20) after locked;</sql>
+  <sql>alter table forumthread add lockDate datetime after lockBy;</sql>
+  <sql>alter table forumthread modify locked varchar(255);</sql>
```

Comment:
This will completely wreck existing database data. When a column is changed from `bit` to `varchar(255)`, existing data will be converted to strings `0` and `1` which are incompatible with the back-end enum values. After making such a drastic change, something like this is also needed:

`update ForumThread set locked=null where locked='0';`
`update ForumThread set locked='ALL' where locked='1';`

Reply:
This suggested modification also wrecks current databases when checking out other branches after this one IF any of the local discussion threads have locked boolean set.